src/Domain/Creators/TimeDependentOptions/BinaryCompactObject.cpp

```diff
@@ -189,102 +206,56 @@ TimeDependentMapOptions<IsCylindrical>::create_functions_of_time(
   // ExpansionMap FunctionOfTime for the function \f$a(t)\f$ in the
   // domain::CoordinateMaps::TimeDependent::RotScaleTrans map
   if (expansion_map_options_.has_value()) {
-    result[expansion_name] =
-        std::make_unique<FunctionsOfTime::PiecewisePolynomial<2>>(
-            initial_time_,
-            std::array<DataVector, 3>{
-                {{gsl::at(expansion_map_options_.value().initial_values, 0)},
-                 {gsl::at(expansion_map_options_.value().initial_values, 1)},
-                 {0.0}}},
-            expiration_times.at(expansion_name));
+    result[expansion_name] = time_dependent_options::get_expansion(
+        expansion_map_options_.value(), initial_time_,
+        expiration_times.at(expansion_name));
 
     // ExpansionMap FunctionOfTime for the function \f$b(t)\f$ in the
     // domain::CoordinateMaps::TimeDependent::RotScaleTrans map
     result[expansion_outer_boundary_name] =
-        std::make_unique<FunctionsOfTime::FixedSpeedCubic>(
-            1.0, initial_time_,
-            expansion_map_options_.value().outer_boundary_velocity,
-            expansion_map_options_.value().outer_boundary_decay_time);
+        time_dependent_options::get_expansion_outer_boundary(
+            expansion_map_options_.value(), initial_time_);
   }
 
   // RotationMap FunctionOfTime for the rotation angles about each
-  // axis.  The initial rotation angles don't matter as we never
-  // actually use the angles themselves. We only use their derivatives
-  // (omega) to determine map parameters. In theory we could determine
-  // each initial angle from the input axis-angle representation, but
-  // we don't need to.
```

Comment:
Could you add some docs to `QuaternionFunctionOfTime` so we don't get confused by this again?